### PR TITLE
adds new content section

### PR DESCRIPTION
**Pull request by andrew-1234**
- software tips

### CODE REVIEW

File: content/resources/help-centre/software-tips/command-line.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+text file. 
+
+The CLI is also known as a **CLI shell**. Sometimes simply referred to as a
+**shell**. A shell is essentially a program that we open on our computer, in
+order to type commands that we want the computer to execute. There are different
+'shells' available, depending on your operating system. You have probably seen
+or heard about `cmd.exe` or command prompt, which is a type of shell which comes
+with Windows by default. But there are other shell programs we can download and
+use, which have some added functionalities. For Windows we recommend downloading
+and using the following:
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/command-line.md:20]: This needs a little bit of work. Some of the definitions are not quite right. Read https://www.hanselman.com/blog/whats-the-difference-between-a-console-a-terminal-and-a-shell. 

We might need another way to present this document (a different title perhaps). The short of it is:

- CLI: command line interface - a text-based way of running programs (rather than graphical)
- shell: a program that a user sends input to (can be graphical, (Windows Explorer) or textual (MS-DOS))

File: content/resources/help-centre/software-tips/command-line.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+- [Windows 
+    Terminal](https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK701?hl=en-au&gl=au)
+- [Powershell 
+  7](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.2)
+  - Note: PowerShell 7 is different to PowerShell 5, which is likely already
+   installed on your system by default. 
+
+If you are using Linux or macOS, then you can use the terminal, which is
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/command-line.md:29]: we'd probably add a note that Bash is the default shell for those applications

File: content/resources/help-centre/software-tips/command-line.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+included by default for those systems. PowerShell 7 is also available for both
+[Linux](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-linux?view=powershell-7.2)
+and
+[macOS](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-macos?view=powershell-7.2).
+
+
+## How do I use it?
+
+{{% include "resources/lessons/shared/basic-terminal-use.md" %}}
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/command-line.md:38]: Nice :-) I'm glad you included that.

File: content/resources/help-centre/software-tips/github.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+title: GitHub
+---
+
+## What is it?
+
+[GitHub](https://github.com/) is a website and service which lets people store
+their code in the cloud, with a generous free option that has all the features
+you will need to get started. In simple terms, you can think of it as being like
+a DropBox for developers. But more than being just a place to store code, the
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/github.md:10]:
```suggestion
a DropBox for anyone who writes code. But more than being just a place to store code, the
```

File: content/resources/help-centre/software-tips/github.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+value of GitHub lies in its version control capabilities, and open source
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/github.md:11]: version control: in the scientific context allows for reproducibility and sovereignty of IP (you can see the history of an idea develop).

File: content/resources/help-centre/software-tips/github.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+collaborative community. 
+
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/github.md:13]: github also has tight integrations with zenodo which will allow researchers to mint DOIs for the software outputs

File: content/resources/help-centre/software-tips/github.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+## When to use it
+
+Whether you are writing code by yourself, within a small team, or even as part
+of a large collaboration, GitHub can help you to safely and securely manage your
+projects, and work together with others from all over the world.
+
+{{% hint info %}}
+**Fun fact:**  
+The Open Ecoacoustics website is hosted as a [repository](https://github.com/ecoacoustics/website) on GitHub!
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/github.md:22]:
```suggestion
This very website is hosted as a [repository](https://github.com/ecoacoustics/website) on GitHub!

Anyone can contribute!
```

File: content/resources/help-centre/software-tips/github.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+start is the [Hello World](https://docs.github.com/en/get-started/quickstart/hello-world) quick start
+guide provided by GitHub. By following this short tutorial, you will have learnt
+a number of skills such as:
+
+- How to make an open source repository (a place where you can store your project)
+- How to make a branch (a type of isolated version of your project where you can
+  make changes without affecting the main project)
+- How to make changes to a file, and "commit" those changes to GitHub
+- How to open and merge a pull request - which pulls the content from the branch
+  you created, into your main project!
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/github.md:37]: also link to the software carpentry courses.

perhaps also suggest using GitHub Desktop as an easy way to get started

File: content/resources/help-centre/software-tips/package-manager.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+ref "command-line.md" %}}).
+
+## Get started
+
+The type of package manager you use will depend on your operating system. We
+recommend three main options: 
+
+- [Chocolatey](https://chocolatey.org/) on Windows
+- [Homebrew](https://brew.sh/) on macOS 
+- [Snap](https://snapcraft.io/) on Linux
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/package-manager.md:42]: we probably shouldn't recommend snap (as much as I want snap to work there are some larger issues it is facing at the moment)... maybe rephrase to see that each variant of linux has its own package manager. apt, yum, snap etc..

File: content/resources/help-centre/software-tips/package-manager.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+title: Package manager
+---
+
+## What is a package manager?
+
+Much of the software that is mentioned on this website includes package manager
+installation instructions, if they are available. But what is a package manager?
+Essentially, it is a piece of software that lets us easily and conveniently
+download and install applications (known as packages), and keep those packages
+up to date. A package manager runs in the [command line]({{% ref
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/package-manager.md:11]: not strictly true; windows store is a package manager (and the new `winget` actually uses the same backend as the store) and various linux package managers have graphical UIs (like snap on linux)

File: content/resources/help-centre/software-tips/package-manager.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+```ps 
+> choco install ffmpeg
+```
+
+That means less time fussing around with installations, and more time for you to
+focus on the things that matter. Besides FFmpeg, there are many useful software
+packages that are commonly used for ecoacoustics which can be installed with
+package managers.
+
+The other great thing about package managers is that they take care of your
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/package-manager.md:30]: not always, use looser language

File: content/resources/help-centre/software-tips/package-manager.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+## Get started
+
+The type of package manager you use will depend on your operating system. We
+recommend three main options: 
+
+- [Chocolatey](https://chocolatey.org/) on Windows
+- [Homebrew](https://brew.sh/) on macOS 
+- [Snap](https://snapcraft.io/) on Linux
+
+### How to install
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/package-manager.md:44]: seems like this choco guide (and a brew guide) should be a sub-pages

i see the tabs... but we're not really showing alternate functionality...

File: content/resources/help-centre/software-tips/path.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+title: The PATH variable
+---
+
+## What is the "PATH"?
+
+If you have been following the lessons and help guides available on this
+website, and installing software such as FFmpeg and AP (AnalysisPrograms) then
+you have probably seen the term `PATH` by now. 
+
+The `PATH` variable is a system wide variable (known as en environment variable)
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:11]: Not strictly true. 

Windows has two path environment variables that get merged: a system one and a per-user one.

```suggestion
The `PATH` variable an environment variable. Environment variables are values that 
```

File: content/resources/help-centre/software-tips/path.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+that any program can access. The `PATH` variable contains a list of folders that
+your computer can check when it searches for programs to run.
+
+In simple terms, if we want to use a command from a particular software in our
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:15]:
```suggestion
In simple terms, if we want to use a program in our
```

File: content/resources/help-centre/software-tips/path.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+shell (see the [command line]({{% ref "command-line.md" %}}) guide), we have two
+choices. We can specify the full path to the `.exe` file, and then write our
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:17]:
```suggestion
choices. We can specify the full path to the program (often referred to as a binary,  usually has an extension like `.exe` on Windows) file, and then write our
```

File: content/resources/help-centre/software-tips/path.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+code:
+
+```ps
+> C:\Users\Anthony\AP\AnalysisPrograms.exe 
+```
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:22]: since AP has a linux installer you can show the linux variant here too

File: content/resources/help-centre/software-tips/command-line.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+**shell**. A shell is essentially a program that we open on our computer, in
+order to type commands that we want the computer to execute. There are different
+'shells' available, depending on your operating system. You have probably seen
+or heard about `cmd.exe` or command prompt, which is a type of shell which comes
+with Windows by default. But there are other shell programs we can download and
+use, which have some added functionalities. For Windows we recommend downloading
+and using the following:
+
+- [Windows 
+    Terminal](https://apps.microsoft.com/store/detail/windows-terminal/9N0DX20HK701?hl=en-au&gl=au)
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/command-line.md:23]: we should strongly recommend windows terminal - tabs, speed, zooming, themes, profiles, split panes, accuracy, colors, and most importantly emoji

Comment (atruskie) [reply content/resources/help-centre/software-tips/command-line.md:23]: i sent you screen shots that show some differences

File: content/resources/help-centre/software-tips/package-manager.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+
+{{% tab "Chocolatey" %}} 
+**Install Chocolately for Windows:**
+1. Visit the [installing chocolatey](https://community.chocolatey.org/courses/installation/installing?method=installing-chocolatey) 
+page
+2. If you have a preference, choose from the installation methods listed
+3. If you are unsure, just click "Basic Chocolatey Install"
+4. Open an administrative shell (Command Prompt)
+    - Click the Start button, and search for "cmd" or "command prompt", 
+    - right-click "command prompt" and select "Run as Administrator"
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/package-manager.md:56]:
```suggestion
    - right-click "command prompt" and select _Run as Administrator_
```

I tend to always italicize labels that need to be clicked on. Looks cleaner IMO

File: content/resources/help-centre/software-tips/package-manager.md [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+5. Paste the installation command listed on the `Install with cmd.exe` section
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/package-manager.md:57]:
```suggestion
5. Paste the installation command listed on the _Install with cmd.exe_ section
```

File: content/resources/help-centre/software-tips/path.md [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+not. Open up a shell (such as PowerShell on Windows, or terminal on
+Linux/macOS), and type a command for the program we are interested in testing. 
+
+```ps
+> FFmpeg
+```
+
+If the command successfully returns some kind of output about the package, then
+it is available on your `PATH`.  
+
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:44]: you probably want to show them how to see path, 

`printenv | grep PATH` in bash and ` echo $env:Path` in pwsh

File: content/resources/help-centre/software-tips/path.md [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+## Adding to PATH
+
+If you use a [package manager]({{% ref "package-manager.md" %}}) such as
+[chocolatey](https://chocolatey.org/) on Windows, [homebrew](https://brew.sh/)
+on macOS or [snap](https://snapcraft.io/) on Linux, then packages you install
+with those commands should be automatically added to your `PATH` variable. This
+saves you having to manually change the `PATH` variable. The automatic installer
+for [AP](https://ap.qut.ecoacoustics.info/basics/installing.html?tabs=windows)
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:52]:
```suggestion
for [AP](https://ap.qut.ecoacoustics.info/basics/installing.html)
```

File: content/resources/help-centre/software-tips/path.md [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+3. Add the following line to the end:
+
+`PATH=$PATH:<REPLACE-ME>`
+
+where you replace the `<REPLACE-ME>` with the directory from step 1.
+
+4. Close and save the file
+5. Run the same command in your current shell (PATH=$PATH:<REPLACE-ME>) to see the change take effect immediately (or restart your shell).
+{{% /tab %}}
+{{% /tabs %}}
```
Comment (atruskie) [inline content/resources/help-centre/software-tips/path.md:95]: so fun fact: there are system wide environment files you can configure for linux (which would be the same as your windows example)
image


and in powershell you can do `$env:Path=$env:Path;<REPLACE-ME>` in your powershell profile as well!

File: content/resources/help-centre/software-tips/autohotkey.md [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+### First use
+
+Right click on the downloaded AHK file, and `Open with` a text editor such as
+Notepad. Have a read through the information in the file.  In particular, check
+to make sure that the following line of code correctly specifies your Raven
+version number:
+
+`#IfWinActive, Raven Pro 1.6.1`
+
+If you are using a different version of Raven, you will have to change this
```
[the remark this answers is not in the document]
Comment (andrew-1234) [reply content/resources/help-centre/software-tips/autohotkey.md:41]: good point - I found [SetTitleMatchMode](https://www.autohotkey.com/docs/commands/SetTitleMatchMode.htm) and added it to the script, so this explanation is no longer necessary. It should match Raven Pro win title no matter what version.

File: content/resources/help-centre/software/path.md [hunk 24]
```diff
@@ -0,0 +1,113 @@
...
+
+```ps
+> FFmpeg
+```
+
+If the command successfully returns some kind of output about the package, then
+it is available on your `PATH`.  
+
+You can check the list of folders contained in your `PATH` variable
+using `printenv | grep PATH` in Bash, `echo $env:Path` in PowerShell, and
```
Comment (atruskie) [inline content/resources/help-centre/software/path.md:53]:
```suggestion
using `printenv PATH` in Bash, `echo $env:Path` in PowerShell, and 
```

File: content/resources/help-centre/software/path.md [hunk 25]
```diff
@@ -0,0 +1,113 @@
...
+```ps
+> FFmpeg
+```
+
+If the command successfully returns some kind of output about the package, then
+it is available on your `PATH`.  
+
+You can check the list of folders contained in your `PATH` variable
+using `printenv | grep PATH` in Bash, `echo $env:Path` in PowerShell, and 
+`echo$env:$path` in Zsh. If the folder where you installed a program you want to
```
Comment (atruskie) [inline content/resources/help-centre/software/path.md:54]:
```suggestion
`printenv PATH` in Zsh. If the folder where you installed a program you want to
```

**Review by atruskie — APPROVED** [review]
